### PR TITLE
A new symmetric stagger keyboard: Angel.

### PR DESCRIPTION
**Pull request by jamessa**
## Description

A new symmetric staggered keyboard: Angel. >

## Types of Changes

- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jamessa** [comment]
I think Travis build status is wrong. In fact it failed on docker image download. 

docker: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit.

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by jamessa** [comment]
It looks like force push will affect review status. I will freeze current layout for review. Thank you all for making such great QMK that I can design the keyboard interaction instantaneously. Amazing.

File: keyboards/angel/config.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
+/* Bootmagic Lite key configuration */
+// #define BOOTMAGIC_LITE_ROW 0
+// #define BOOTMAGIC_LITE_COLUMN 0
```
Comment (fauxpark) [inline keyboards/angel/config.h:112]:
```suggestion
```

File: keyboards/angel/info.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "keyboard_name": "Angel",
+    "keyboard_folder": "angel",
+    "url": "https://github.com/jamessa/Angel",
+    "maintainer": "jamessa",
+    "width": 13,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 47,
```
Comment (fauxpark) [inline keyboards/angel/info.json:10]:
```suggestion
```

File: keyboards/angel/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        UNDO,    CUT,     COPY,    PSTE,    _______,    _______,    _______,    _______, KC_LEFT, KC_DOWN, KC_RGHT, _______,
+        REDO,    WKSPLF,  _MSNCTL, WKSPRT,  _______, _______,TO(_KAI), _______, _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END,
+                 _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+    )
+};
+
+void keyboard_post_init_user(void) {
+  debug_enable=true;
+  debug_matrix=true;
+}
```
Comment (fauxpark) [inline keyboards/angel/keymaps/default/keymap.c:79]:
```suggestion
```

File: keyboards/angel/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+To build the firmware.
+
+    qmk compile -kb angel -km default
+or
+
+    make angel:default
+
+To flash the Pro Micro board, short `RESET` pin and run following command.
+
+     avrdude -p atmega32u4 -c avr109 -U flash:w:angel_default.hex:i -P /dev/tty.usbmodem14201
```
Comment (fauxpark) [inline keyboards/angel/readme.md:20]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make angel:default

Flashing example for this keyboard:

    make angel:default:flash
```

File: keyboards/angel/rules.mk [hunk 5]
```diff
@@ -0,0 +1,10 @@
+MCU = atmega32u4
+BOOTLOADER = caterina
+
+# Build Options
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
```
Comment (fauxpark) [inline keyboards/angel/rules.mk:10]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
This should probably go in the `handwired` folder.

**Comment by jamessa** [comment]
Definitely. Let me fix it.

**Pull request merged**